src/gax/src/retry_loop_internal.rs

```diff
@@ -76,29 +97,26 @@ where
                     .lock()
                     .expect("retry throttler lock is poisoned")
                     .on_retry_failure(&flow);
-                on_error(&sleep, flow, delay).await?;
+                match flow {
+                    LoopState::Permanent(e) | LoopState::Exhausted(e) => return Err(e),
+                    LoopState::Continue(e) => {
+                        // TODO(#2307) - For a stricter limit, we should query
+                        //               the retry policy to determine the
+                        //               remaining time.
+                        // let remaining_time = retry_policy.remaining_time(loop_start, attempt_count);
```

Comment:
or you could could move the sleep to the beginning of the loop, and just not do it for the first request? Your new `Attempt` enum should make that much easier.